### PR TITLE
Makefile: install buildx from binary release, instead of building

### PR DESCRIPTION
**Pull request by thaJeztah**
This was originally added in 833444c0d605b0cf86bc9dbdb436601434fd3493 (https://github.com/moby/moby/pull/39340), at which time buildx did not yet have a release, so we had to build from source.

Now that buildx has binary releases on GitHub, we should be able to consume those binaries instead of building.

### CODE REVIEW

**Comment by thaJeztah** [comment]
Ah, interesting; https://ci-next.docker.com/public/blue/rest/organizations/jenkins/pipelines/moby/branches/PR-41949/runs/5/nodes/337/log/?start=0

related bundles; https://ci-next.docker.com/public/job/moby/job/PR-41949/5/artifact/unit-bundles.tar.gz

```
=== FAIL: daemon/logger TestCopierWithSized/With_RingLogger (0.01s)
    --- FAIL: TestCopierWithSized/With_RingLogger (0.01s)
        copier_test.go:265: invalid character '\x00' looking for beginning of value

=== FAIL: daemon/logger TestCopierWithSized (0.01s)
```

File: Makefile [hunk 1]
```diff
@@ -273,22 +275,9 @@ buildx: bundles/buildx ## build buildx cli tool
...
-	if [ -n "$${id}" ]; then \
-		docker cp $${id}:/usr/bin/buildx $@ \
-		&& touch $@; \
-		docker rm -f $${id}; \
-	fi
-
-	$@ version
+	goarch="$$(uname -m | sed -e 's/x86_64/amd64/g')"; \
+	goos="$$(uname -s | awk '{print tolower($$0)}')"; \
+	curl -o $@ -fsSL "https://github.com/docker/buildx/releases/download/${BUILDX_VERSION}/buildx-${BUILDX_VERSION}.$${goos}-$${goarch}" \
```
Comment (thaJeztah) [inline Makefile:281]: Discussing with @tonistiigi @cpuguy83 and opting to not do the automatic detection, but to default to ppc64le (which is what we use it for), and a make variable to override.

Comment (thaJeztah) [reply Makefile:281]: We can use the script from https://github.com/moby/buildkit/blob/master/hack/install-buildx (needs some minor changes to allow overriding the parameters)

**Review by AkihiroSuda — APPROVED** [review]

**Comment by thaJeztah** [comment]
I'll be updating this with https://github.com/moby/moby/pull/41949#discussion_r566373194. PR is not "urgent", so will give a ping once that's done.

File: Makefile [hunk 2]
```diff
@@ -273,22 +275,6 @@ buildx: bundles/buildx ## build buildx cli tool
...
-		.
-
-	id=$$(docker create moby-buildx:$${BUILDX_COMMIT:-latest}); \
-	if [ -n "$${id}" ]; then \
-		docker cp $${id}:/usr/bin/buildx $@ \
-		&& touch $@; \
-		docker rm -f $${id}; \
-	fi
-
+	curl -fsSL https://raw.githubusercontent.com/thaJeztah/buildkit/6d75906c685f2a612ee3f72ee3b4bacc8948cd0b/hack/install-buildx | VERSION="$(BUILDX_VERSION)" BINDIR="$(@D)" bash
```
Comment (thaJeztah) [inline Makefile:279]: Temporarily using my branch from https://github.com/moby/buildkit/pull/2106

Comment (thaJeztah) [reply Makefile]: https://github.com/moby/buildkit/pull/2106 was merged 👍

**Comment by thaJeztah** [comment]
@tonistiigi @tianon PTAL

**Comment by thaJeztah** [comment]
@cpuguy83 PTAL 🤗

**Review by tonistiigi — APPROVED** [review]

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Comment by cpuguy83** [comment]
We should validate the SHA's for this binary, I suppose.

**Comment by thaJeztah** [comment]
We'd need a list of sha's (one per os/arch)

**Pull request merged**